### PR TITLE
Update plugin manifest doc

### PR DESCRIPTION
**Pull request by astraw99**
To make the plugin manifest doc better to new developers.

### CODE REVIEW

File: site/content/docs/developer-guide/plugin-manifest.md [hunk 1]
```diff
@@ -99,6 +100,8 @@ Krew makes it possible to install the same plugin on different operating systems
 (e.g., `windows`, `darwin` (macOS), and `linux`) and different architectures
 (e.g., `amd64`, `386`, `arm`, `arm64` and `ppc64le`).
 
+See [all supported platforms](https://github.com/kubernetes-sigs/krew/blob/master/cmd/validate-krew-manifest/main.go#L233-L249).
```
Comment (ahmetb) [inline site/content/docs/developer-guide/plugin-manifest.md:103]: This link will almost certainly go out of date. :)

I think the right approach is to point to krew repo's releases page and say "Only plugins in platforms that krew itself is distributed are supported.", right?

File: site/content/docs/developer-guide/plugin-manifest.md [hunk 2]
```diff
@@ -99,6 +100,9 @@ Krew makes it possible to install the same plugin on different operating systems
 (e.g., `windows`, `darwin` (macOS), and `linux`) and different architectures
 (e.g., `amd64`, `386`, `arm`, `arm64` and `ppc64le`).
 
+See [all supported platforms](https://github.com/kubernetes-sigs/krew/releases).
+Only plugins in platforms that krew itself is distributed (Release Assets) are supported.
```
Comment (chriskim06) [inline site/content/docs/developer-guide/plugin-manifest.md:104]: also wanted to float the idea of having a static list here. i know its not great to have to update in multiple places when something new is supported but this isnt a frequent occurrence. when ppc64le was added the contributor also [remembered to update the docs](https://github.com/kubernetes-sigs/krew/pull/755/files#diff-bbfd929e7884a7e220b4bcddf4c4f992cde11c8b31fab08f874f7caf9ec87fa1R100). an option to help any drift between what's supported and the docs would be to add a comment (maybe to [here](https://github.com/kubernetes-sigs/krew/blob/master/hack/krew.yaml#L28)) as a reminder to update the docs area with the new platform.

if not then i think it reads better as part of the previous paragraph like
```suggestion
The supported platforms for plugins are the ones that Krew are distributed in.
See all supported platforms on the [releases page](https://github.com/kubernetes-sigs/krew/releases).
```

Comment (astraw99) [reply site/content/docs/developer-guide/plugin-manifest.md]: Got it, updated.
PTAL thanks.

**Comment by astraw99** [comment]
/ping @ahmetb @chriskim06 
PTAL thanks.

**Comment by ahmetb** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**